pkg/kubelet/cm/cpumanager/policy_static.go

```diff
@@ -621,11 +621,10 @@ func (p *staticPolicy) generateCPUTopologyHints(availableCPUs cpuset.CPUSet, reu
 		// numa node bitmask to satisfy the CPU request.
 		numMatching := 0
 		for _, c := range reusableCPUs.List() {
-			// Disregard this mask if its NUMANode isn't part of it.
-			if !mask.IsSet(p.topology.CPUDetails[c].NUMANodeID) {
-				return
+			// Still allow this mask if reusableCPU's NUMANode isn't part of it.
+			if mask.IsSet(p.topology.CPUDetails[c].NUMANodeID) {
+				numMatching++
```

Comment:
I don't see anything init-container specific in this code. Does it mean that it will work for all containers in the same pod?